File: product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> answer(nums.size(), 1);
        int temp1 = 1;
        int temp2 = 1;
        int n = nums.size() - 2;
        for (int i = 1; i<nums.size(); i++){
            temp1 *= nums[i-1];
            answer[i] *= temp1;
            temp2 *= nums[n+1];
            answer[n] *= temp2;
            n--;
        }
        return answer;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answers = [1]
        temp = 1

        for i in range(1, len(nums)):
            temp *= nums[i-1]
            answers.append(temp)
        
        temp = 1
        for i in range(len(nums)-2, -1, -1):
            temp *= nums[i+1]
            answers[i] *= temp
        
        return answers

